### DISCUSSION TITLE
Regarding the bmi2 binary for Stockfish 16

### DISCUSSION THREAD

**Discussion by LunaticBFF57** [official-stockfish/Stockfish#4655] [Q&A] [answered]
Hi Stockfish community, I saw the release files for Stockfish 16 and saw that the BMI2 binary that was existent for 15.1 was not there. Why is that?

**Comment by dav1312** [comment, marked as answer]
The binaries are generated automatically and BMI2 was not in the list of binaries that are created:
https://github.com/official-stockfish/Stockfish/blob/9a2d50ecccfc737249245280586924ee3ef53abb/.github/workflows/stockfish_binaries.yml#L41-L47

Now we have BMI2:
https://github.com/official-stockfish/Stockfish/blob/e87e103ca994570d42f30f61f923986656a5df14/.github/workflows/stockfish_binaries.yml#L41-L50

You can now download BMI2 binaries of our latest pre-release: https://github.com/official-stockfish/Stockfish/releases?q=prerelease%3Atrue

**Reply by LunaticBFF57** [reply]
Thanks, mate!

**Reply by LunaticBFF57** [reply]
Also, is it possible for the devs to add the bmi2 binary to the release of Stockfish 16 (the "latest" version, not the pre-release version)?

**Reply by dav1312** [reply]
_Technically_ it is possible to edit a release to _manually_ add new binaries, but I'm not sure if it will be done.

**Reply by vondele** [reply]
no, I do not intend to manually upload additional binaries. Various options
* compile from sources
* use a pre-release from our development builds
* look elsewhere for binaries (e.g. abrok)
* ignore the small speed difference for now.

**Reply by LunaticBFF57** [reply]
alright then

**Comment by hekunzai** [comment]
where is  fish 16.0 bmi2?

**Reply by LunaticBFF57** [reply]
It doesn't contain it at all, only 16.1 has it. (If you are looking for only 16.0, there are different binaries are recommended depending on your CPU compatibility.)

**Reply by hekunzai** [reply]
Thank you very much
